app/store/charmstore-api.js

```diff
@@ -143,25 +143,37 @@ YUI.add('charmstore-api', function(Y) {
       @method lowerCaseKeys
       @param {Object} obj The source object with the uppercase keys.
       @param {Object} host The host object in which the keys will be assigned.
+      @param {Object} exclude Exclude a particular level from lowercasing when
+        recursing.
       @return {Undefined} Does not return a value, modifies the supplied host
         object in place.
     */
-    _lowerCaseKeys: function(obj, host) {
+    _lowerCaseKeys: function(obj, host, exclude) {
       if (!obj) {
         return;
       }
       Object.keys(obj).forEach(function(key) {
-        host[key.toLowerCase()] =
+        // An exclude of 0 means "don't lowercase this level".
```

Comment:
It isn't because that wouldn't be correct. If you pass in an exclude of 3, it would mean "exclude the fourth level down from lowercasing". As the function recurses, it decrements the exclude such that, when it hits the fourth level, `exclude == 0`. But that's from an internal perspective. From an external perspective, you specify 3 and can expect the fourth level to be excluded.